### PR TITLE
INT-4355: Add support for DomPOX WS communication

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4355

**Cherry-pick to 4.3.x**

### CODE REVIEW

**Review by garyrussell — APPROVED** [review]
That's all.

File: spring-integration-ws/src/test/java/org/springframework/integration/ws/SimpleWebServiceOutboundGatewayTests.java [hunk 1]
```diff
@@ -194,6 +194,49 @@ public boolean handleRequest(MessageContext messageContext) throws WebServiceCli
...
+
+		};
+		gateway.setInterceptors(interceptorAdapter);
+		gateway.setMessageFactory(new DomPoxMessageFactory());
+		gateway.afterPropertiesSet();
+
+		String request = "<test>foo</test>";
+		try {
+			gateway.handleMessage(new GenericMessage<>(request));
+		}
```
[the remark this answers is not in the document]
Comment (artembilan) [reply spring-integration-ws/src/test/java/org/springframework/integration/ws/SimpleWebServiceOutboundGatewayTests.java]: ??? It's going to fail anyway, Gary.

That is the point of these tests: to be executed until the network communication. Only what we need to know that we really prepare the correct object to be send over the network

**Comment by garyrussell** [comment]
Merged as 5092d482b525687d28238d0358d28637d2359fb9 cherry-picked as 6193263ab001151c2571f795fd5a976cd5abfdfd

File: spring-integration-ws/src/test/java/org/springframework/integration/ws/SimpleWebServiceOutboundGatewayTests.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (garyrussell) [reply spring-integration-ws/src/test/java/org/springframework/integration/ws/SimpleWebServiceOutboundGatewayTests.java]: Exactly, but a blind catch for `Exception` is dangerous - some unexpected exception could occur; worse, something might change that causes no exception to be thrown which we wouldn't detect; hence we always should add a JUnit `fail()` whenever an exception is expected so the test fails if no exception is thrown.

**Pull request closed without merging**